test/test_katello/test_agent/test_pulp/test_handler.py

```diff
@@ -0,0 +1,30 @@
+import os
+import sys
+sys.path.append(os.path.join(os.path.dirname(__file__), '../../src/'))
+from unittest import TestCase
+
+from katello.agent.pulp import handler
+from katello.agent.pulp import libdnf
```

Comment:
If you want to test only the dnf side of this you can annotate this test class:

```
@unittest.skipIf('dnf' not in sys.modules, "DNF not present")
class PackageHandler(TestCase):
```

If you wanna test both yum and dnf sides from this class, this could work if they have the same API and same expectations:

```
from katello.agent.pulp import handler
try:
  from katello.agent.pulp import libdnf as packagelib
except:
  from katello.agent.pulp import libyum as packagelib
```

and in your test:

```
@patch('packagelib.Package.update')
def test_update
```

```